Notebooks/DC2Cookbook.ipynb

```diff
@@ -0,0 +1,559 @@
+    "\n",
+    "<a id='toc'></a>\n",
+    "This Note is organized as follows. We first provide an [overview](#overview) of the DRP pipeline, summarizing the relevant section in [LDM-151](ls.st/ldm-151). Then, we present the recipes in three sections: \n",
+    "1. [Image Coaddition and Object Detection](#coadds)\n",
+    "2. [Difference Image Analysis and DIASource Detection](#diasources)\n",
+    "3. [DIAObject Generation and Light Curve Forced Photometry](#forcedphot)\n",
+    "\n",
+    "We then provide some brief [concluding remarks](#conclusions).  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='overview'></a>\n",
+    "## Data Release Processing Overview\n",
+    "\n",
+    "The DRP pipelines are summarized in [LDM-151](ls.st/ldm-151) by the following Figure:\n",
+    "\n",
+    "<img src=\"https://github.com/lsst/LDM-151/raw/master/figures/drp_summary.png\" width=40% align=\"left\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Anticipating that difference image analysis (DIA) will not be required in the DC2 main survey area, it makes sense to preserve the DC1 grouping of the DRP pipelines, into static sky and dynamic sky pipelines. \n",
+    "  * The static sky analysis involves image characterization, calibration and coaddition followed by `Object` generation and measurement. \n",
+    "  * The dynamic sky analysis makes use of the calibrated images, but involves 1) the construction of a `TemplateCoadd` image followed by image differencing, `DIASource` detection in the difference images, and then 2) association of those `DIASources` into `DIAObjects` which then form the target positions for forced photometry. Light curves can then be extracted from the `DIAForcedSource` table.\n",
+    "\n",
+    "At the time of writing, multi-epoch object characterization is not yet available in the DM stack. As a result, the `Objects` are the same as the \"Preliminary Objects\", and are made by detecting sources in the `Coadd` images.\n",
+    "\n",
+    "Note that there will be difference image analysis (DIA) carried out in both the Alert Production and Data Release Processing pipelines. The main difference is that the APP will need to build `DIAObjects` on the fly, while the DRP will have all data available to it at once, after the fact. The DRP can make the best possible `TemplateCoadd` reference image, and then generate new `DIASources` and aggregate them into new, \"reprocessed\" `DIAObjects` ready for a single pass of forced photometry at their locations of all these reprocessed `DIAObjects`, which will result in light curves extending over the duration of the whole survey to that point.  As in Twinkles, in DC2 our goal for the dynamic sky analysis is to emulate the production of cosmology-ready LSST supernova and lensed quasar light curves, which is why we retain the focus on the _data release reprocessing_ of the visit images."
```

Comment:
I think we should probably mention forced photometry on per epoch images seeded by the objects detected from the coadd.